### PR TITLE
fix(medusa-plugin-sendgrid): Use correct SendGrid client

### PR DESCRIPTION
**Pull request by olivermrbl**

### CODE REVIEW

**Review by adrien2p — APPROVED** [review]
Lgtm, what the transporter_ was?

**Comment by olivermrbl** [comment]
I am not sure where that came from. Maybe @pevey knows? 

But an oversight from my side when I did the review

**Comment by adrien2p** [comment]
Got it, does that also means that we are missing some tests maybe

**Comment by olivermrbl** [comment]
Yeah, sure does. Pushing to get a new release out first, as the current latest is broken

**Comment by olivermrbl** [comment]
Added tests @adrien2p

File: packages/medusa-plugin-sendgrid/src/services/__tests__/sendgrid.js [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+jest.mock("@sendgrid/mail")
+
+const mockedSendGrid = SendGrid
+mockedSendGrid.setApiKey.mockResolvedValue(mockedSendGrid)
+mockedSendGrid.send.mockResolvedValue(Promise.resolve())
+
+describe("SendGridService", () => {
+  let sendGridService
+
+  const totalsService = {
```
Comment (adrien2p) [inline packages/medusa-plugin-sendgrid/src/services/__tests__/sendgrid.js:14]: comment: we should think at some point to remove deprecated API

File: packages/medusa-plugin-sendgrid/src/services/sendgrid.js [hunk 2]
```diff
@@ -261,10 +261,15 @@ class SendGridService extends NotificationService {
...
-   .then(() => { status = "sent" })
-   .catch((error) => { status = "failed"; console.log(error) })
+    let status
+    await SendGrid.send(sendOptions)
+      .then(() => {
+        status = "sent"
+      })
+      .catch((error) => {
+        status = "failed"
+        console.log(error)
```
Comment (adrien2p) [inline packages/medusa-plugin-sendgrid/src/services/sendgrid.js]: todo: we should use the logger instead of the console API

**Review by adrien2p — APPROVED** [review]
LGTM, one toto

**Comment by pevey** [comment]
I took the error logic from my other notification plugins, and it looks like I failed to change the method name there.

**Pull request merged**